ompi/mca/osc/ucx/osc_ucx_component.c

```diff
@@ -285,7 +285,7 @@ static int component_init(bool enable_progress_threads, bool enable_mpi_threads)
     mca_osc_ucx_component.enable_mpi_threads = enable_mpi_threads;
     mca_osc_ucx_component.wpool = opal_common_ucx_wpool_allocate();
     mca_osc_ucx_component.priority_is_set = false;
-
+    component_register();
```

Comment:
why isn't component_register called without this change? I'd expect MCA framework to call it..

Reply:
I think it's not being called before osc_query as its not a part of the mca lifecycle

```c
ompi_osc_ucx_component_t mca_osc_ucx_component = {
    { /* ompi_osc_base_component_t */
        .osc_version = {
            OMPI_OSC_BASE_VERSION_3_0_0,
            .mca_component_name = "ucx",
            MCA_BASE_MAKE_VERSION(component, OMPI_MAJOR_VERSION, OMPI_MINOR_VERSION,
                                  OMPI_RELEASE_VERSION),
            .mca_open_component = component_open,
            .mca_close_component = component_close,
            .mca_register_component_params = component_register,
        },
        .osc_data = {
            /* The component is not checkpoint ready */
            MCA_BASE_METADATA_PARAM_NONE
        },
        .osc_init = component_init,
        .osc_query = component_query,
        .osc_select = component_select,
        .osc_finalize = component_finalize,
    },
```

Reply:
> It's not being called before osc_query as its not a part of the mca lifecycle
> 
> ```c
> ompi_osc_ucx_component_t mca_osc_ucx_component = {
>     { /* ompi_osc_base_component_t */
>         .osc_version = {
>             OMPI_OSC_BASE_VERSION_3_0_0,
>             .mca_component_name = "ucx",
>             MCA_BASE_MAKE_VERSION(component, OMPI_MAJOR_VERSION, OMPI_MINOR_VERSION,
>                                   OMPI_RELEASE_VERSION),
>             .mca_open_component = component_open,
>             .mca_close_component = component_close,
>             .mca_register_component_params = component_register,
>         },
>         .osc_data = {
>             /* The component is not checkpoint ready */
>             MCA_BASE_METADATA_PARAM_NONE
>         },
>         .osc_init = component_init,
>         .osc_query = component_query,
>         .osc_select = component_select,
>         .osc_finalize = component_finalize,
>     },
> ```

@janjust WDYT?

Reply:
I don't think this call is necessary, we have always been able to set env variables without any issue, not sure what we are trying to solve here by calling it again. @roiedanino can you please clarify further?

Reply:
This PR should solve the segfault from #11974 caused by accessing `tls` and `devices` mca variables while not initialized because `component_register()` is not being called for `osc` components before `osc_query()`, AFAIU osc_query is not part of the mca component lifecycle, therefore, we cannot guarantee variables initialization at that point which leads to segfault.

Reply:
Thanks for the clarification. I think  ```opal_common_ucx_support_level_t()``` should have a check inside to make sure ```tls``` is not NULL before calling ```strcmp()```. If ```tls``` is NULL, then we can call ````opal_common_ucx_mca_var_register(NULL)```` to register it and then continue. In this way, we do not have to rely on the caller of ```opal_common_ucx_support_level_t()``` to worry about ```tls``` not being set. What do you think?

Reply:
Sounds good, will try that
Thanks!